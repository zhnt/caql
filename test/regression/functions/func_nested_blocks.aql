let global_x = 100

function test_nested_blocks() {
    print("=== Function start ===")
    let func_x = 200
    print("Function x: " + string(func_x))
    print("Global x in function: " + string(global_x))
    
    if true {
        print("=== If block start ===")
        let if_x = 300
        print("If x: " + string(if_x))
        print("Function x in if: " + string(func_x))
        print("Global x in if: " + string(global_x))
        
        for i = 1, 3 {
            print("=== For block start ===")
            let for_x = 400 + i
            print("For x: " + string(for_x))
            print("If x in for: " + string(if_x))
            print("Function x in for: " + string(func_x))
            print("Global x in for: " + string(global_x))
        }
        
        print("=== Back to if block ===")
        print("If x after for: " + string(if_x))
    }
    
    print("=== Back to function ===")
    print("Function x after if: " + string(func_x))
}

test_nested_blocks()
print("=== Back to global ===")
print("Global x after function: " + string(global_x))
