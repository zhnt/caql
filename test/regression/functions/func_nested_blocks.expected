=== Function start ===
Function x: 200
Global x in function: 100
=== If block start ===
If x: 300
Function x in if: 200
Global x in if: 100
=== For block start ===
For x: 401
If x in for: 300
Function x in for: 200
Global x in for: 100
=== For block start ===
For x: 402
If x in for: 300
Function x in for: 200
Global x in for: 100
=== Back to if block ===
If x after for: 300
=== Back to function ===
Function x after if: 200
=== Back to global ===
Global x after function: 100
